superset-frontend/packages/superset-ui-core/src/hooks/useTruncation/useChildElementTruncation.ts

```diff
@@ -27,92 +27,68 @@ import { RefObject, useLayoutEffect, useState, useRef } from 'react';
-      scrollWidth,
-      parentElementWidth,
-      plusRefWidth,
-    };
-
-    if (
-      previousEffectInfo.parentElementWidth === parentElementWidth &&
-      previousEffectInfo.scrollWidth === scrollWidth &&
-      previousEffectInfo.plusRefWidth === plusRefWidth
-    ) {
-      return;
-    }
+    const onResize = () => {
+      const currentElement = elementRef.current;
+      if (!currentElement) {
+        return;
+      }
+      const plusRefElement = plusRef.current;
+      const { scrollWidth, clientWidth, childNodes } = currentElement;
 
-    if (scrollWidth > clientWidth) {
-      // "..." is around 6px wide
-      const truncationWidth = 6;
-      const plusSize = plusRefElement?.offsetWidth || 0;
-      const maxWidth = clientWidth - truncationWidth;
-      const elementsCount = childNodes.length;
+      if (scrollWidth > clientWidth) {
+        // "..." is around 6px wide
+        const truncationWidth = 6;
+        const plusSize = plusRefElement?.offsetWidth || 0;
+        const maxWidth = clientWidth - truncationWidth;
+        const elementsCount = childNodes.length;
 
-      let width = 0;
-      let hiddenElements = 0;
-      for (let i = 0; i < elementsCount; i += 1) {
-        const itemWidth = (childNodes[i] as HTMLElement).offsetWidth;
-        const remainingWidth = maxWidth - truncationWidth - width - plusSize;
+        let width = 0;
+        let hiddenElements = 0;
+        for (let i = 0; i < elementsCount; i += 1) {
+          const itemWidth = (childNodes[i] as HTMLElement).offsetWidth;
+          const remainingWidth = maxWidth - width - plusSize;
 
-        // assures it shows +{number} only when the item is not visible
-        if (remainingWidth <= 0) {
-          hiddenElements += 1;
+          // assures it shows +{number} only when the item is not visible
+          if (remainingWidth <= 0) {
+            hiddenElements += 1;
+          }
+          width += itemWidth;
         }
-        width += itemWidth;
-      }
 
-      if (elementsCount > 1 && hiddenElements) {
-        setHasHiddenElements(true);
-        setElementsTruncated(hiddenElements);
+        if (elementsCount > 1 && hiddenElements) {
+          setHasHiddenElements(true);
+          setElementsTruncated(hiddenElements);
+        } else {
+          setHasHiddenElements(false);
+          setElementsTruncated(1);
+        }
       } else {
         setHasHiddenElements(false);
-        setElementsTruncated(1);
+        setElementsTruncated(0);
       }
-    } else {
-      setHasHiddenElements(false);
-      setElementsTruncated(0);
+    };
+    const obs = new ResizeObserver(onResize);
+
+    const element = elementRef.current?.parentElement;
+    if (element) {
+      obs.observe(element);
     }
-  }, [
-    elementRef.current?.offsetWidth,
-    elementRef.current?.clientWidth,
-    elementRef,
-  ]);
 
-  return [elementsTruncated, hasHiddenElements];
+    onResize();
+
+    return () => {
+      obs.disconnect();
+    };
```

Comment:
Oh good point! Though to be fair, I’m not sure about readability - I was halfway through writing a comment about how we should return a cleanup function rather than returning `obs.disconnect()`… but it’s early morning, so maybe it’s my fault 😉

Reply:
I'm almost 50-50 here, but I'm leaning in the direction of @betodealmeida here, so I'd say +1 for changing to `return obs.disconnect;`

Reply:
2 vs 1, request for change has passed 🙂

Reply:
@villebro @betodealmeida With `return obs.disconnect`, the page is crashing with `illegal invocation` error. To be honest I don't know why it doesn't work, but it does work with `return () => { obs.disconnect(); }` so I'm going with that

<img width="1728" alt="image" src="https://github.com/apache/superset/assets/15073128/e1c783f6-056c-4075-a695-1a651c6a280c">